src/Dialect/ONNX/ElementsAttr/ElementsAttrBuilder.cpp

```diff
@@ -497,7 +499,273 @@ bool isIdentityPermutation(ArrayRef<uint64_t> perm) {
+batch_axis=1 ( it will be input[1] dimsize for batch_axis=0) for idx on input
+    begin
+      if( batch_axis==1 and idx[1] == i and idx[0] < sequence_lens[i] )
+        dstSrcPositionPairs.push(idx.pos,0) // the destination pos which will be
+        replaced is added, the source postion form where it will be replaced
+        will be computed later
+        timeAxisPosList1[idx[0]].push(idx.pos) // Add this pos to
+        the correspoding timeAxis list.
+      else if ( batch_axis==0 and idx[0] == i and idx[1] < sequence_lens[i] )
+        dstSrcPositionPairs.push(idx.pos,0)
+        timeAxisPosList1[idx[1]].push(idx.pos)
+    end
+
+list<list<position>> timeAxisPosList2 with size input[0] dimsize for
+batch_axis=1 ( it will be input[1] dimsize for batch_axis=0) for idx on input
+    begin
+      if( batch_axis==1 and idx[1] == i and idx[0] < sequence_lens[i] )
+        timeAxisPosList2[idx[0]].push(idx.pos)
+        positionWithinList = timeAxisPosList2[idx[0]].size()
+
+        listAsPerRevSeq = timeAxisPosList1.size()-idx[0]-1
+        sourcePosition = timeAxisPosList1[listAsPerRevSeq][positionWithinList]
+        update the pair in dstSrcPositionPairs for idx.pos with sourcePosition
+      else if( batch_axis==0 and idx[0] == i and idx[1] < sequence_lens[i] )
+        timeAxisPosList2[idx[1]].push(idx.pos)
+        positionWithinList = timeAxisPosList2[idx[1]].size()
+
+        listAsPerRevSeq = timeAxisPosList1.size()-idx[1]-1
+        sourcePosition = timeAxisPosList1[listAsPerRevSeq][positionWithinList]
+        update the pair in dstSrcPositionPairs for idx.pos with sourcePosition
+    end
+  get iterator for dstSrcPositionPairs.
+  for idx on input
+    begin
+      continue till the idx.pos equals iterator.destination
+      update the result[idx.pos] with value at iterator.source
+      increment the iterator
+    end
+*/
+ElementsAttr ElementsAttrBuilder::reverseSequence(ElementsAttr input,
+    ElementsAttr sequenceLength, uint64_t batchAxis, uint64_t timeAxis) {
+
+  ShapedType inputType = input.getShapedType();
+  ArrayRef<int64_t> inputShape = inputType.getShape();
+
+  SmallVector<int64_t> inputStrides;
+  ArrayBuffer<WideNum> inputNums =
+      getWideNumsAndExpandedStrides(input, inputShape, inputStrides);
+
+  SmallVector<int64_t> seqLengthStrides;
+  ArrayRef<int64_t> seqLengthShape = sequenceLength.getShapedType().getShape();
+  ArrayBuffer<WideNum> seqLengthNums = getWideNumsAndExpandedStrides(
+      sequenceLength, seqLengthShape, seqLengthStrides);
 
+  Type elementType = inputType.getElementType();
+
+  return fromWideNums(inputType, [&](MutableArrayRef<WideNum> dstNums) {
+    wideZeroDispatchNonBool(elementType, [&](auto wideZero) {
+      using cpptype = decltype(wideZero);
+      constexpr BType TAG = toBType<cpptype>;
+      // This loop copies each element in  the input to the dstNums
+      for (auto &idxoffs : StridesRange<1>(inputShape, {inputStrides})) {
```

Comment:
Multiple variables in this function could be const, especially for references I think its important to communicate that the referenced object does not get modified
```suggestion
      for (const auto &idxoffs : StridesRange<1>(inputShape, {inputStrides})) {
```

Reply:
Thats correct. It should be const. Changed it.